internal/controller/serverset/serverset.go

```diff
@@ -540,42 +646,40 @@ func (c *external) areNicsUpToDate(ctx context.Context, cr *v1alpha1.ServerSet)
 	return true, nil
 }
 
-func (c *external) ensureNIC(ctx context.Context, cr *v1alpha1.ServerSet, serverID, lanName string, idx int) error {
+func (c *external) ensureNIC(ctx context.Context, cr *v1alpha1.ServerSet, serverID, lanName string, replicaIndex, version int) error {
 	// get the network
 	resourceType := "nic"
-	nicName := getNameFromIndex(cr.Name, resourceType, idx)
-	network := v1alpha1.Lan{}
-	if err := c.kube.Get(ctx, types.NamespacedName{
-		Namespace: cr.GetNamespace(),
-		Name:      lanName,
-	}, &network); err != nil {
-		return err
-	}
 
-	lanID := network.Status.AtProvider.LanID
-	observedNic := v1alpha1.Nic{}
-	err := c.kube.Get(ctx, types.NamespacedName{
-		Namespace: cr.GetNamespace(),
-		Name:      nicName,
-	}, &observedNic)
-	if err != nil && !apiErrors.IsNotFound(err) {
+	res := &v1alpha1.NicList{}
+	err := kube.ListResourceFromLabelIndex(ctx, c.kube, resourceType, replicaIndex, res)
+	if err != nil {
 		return err
 	}
-	// no NIC found, create one
-	if apiErrors.IsNotFound(err) {
-		c.log.Info("Creating NIC", "name", nicName)
-		createNic := fromServerSetToNic(cr, nicName, serverID, lanID)
-		createNic.SetProviderConfigReference(cr.Spec.ProviderConfigReference)
-		return c.kube.Create(ctx, &createNic)
-	}
-
-	// NIC found, check if it's attached to the server
-	if !strings.EqualFold(observedNic.Status.AtProvider.State, ionoscloud.Available) {
-		return fmt.Errorf("observedNic %s got state %s but expected %s", observedNic.GetName(), observedNic.Status.AtProvider.State, ionoscloud.Available)
+	if len(res.Items) == 0 {
+		return c.createNic(ctx, cr, serverID, lanName, replicaIndex, version)
+
+		// network := v1alpha1.Lan{}
+		// if err := c.kube.Get(ctx, types.NamespacedName{
+		// 	Namespace: cr.GetNamespace(),
+		// 	Name:      lanName,
+		// }, &network); err != nil {
+		// 	return err
+		// }
+		// lanID := network.Status.AtProvider.LanID
+		// nicName := kube.GetNameFromIndex(cr.Name, resourceType, replicaIndex, version)
+		// // no NIC found, create one
+		// c.log.Info("Creating NIC", "name", nicName)
+		// createNic := fromServerSetToNic(cr, nicName, serverID, lanID, replicaIndex, version)
+		// createNic.SetProviderConfigReference(cr.Spec.ProviderConfigReference)
+		// return c.kube.Create(ctx, &createNic)
+	} else {
```

Comment:
as our linter recommends drop the `else` case, as your `len(res.Items)==0` ends with `return`